homeassistant/components/conversation/chat_log.py

```diff
@@ -419,3 +419,24 @@ async def async_update_llm_data(
                 "tools": self.llm_api.tools if self.llm_api else None,
             },
         )
+
+    def trim_history(
```

Comment:
This method seems like it's doing the opposite from the PR description

> keeps trimming message counts in the integration, ensuring that ollama only trims what it sends to its own model while preserving global history with other agents

Reply:
Yes, this should not have been included, let me revert.